services:
  syncthing:
    image: lscr.io/linuxserver/syncthing:1.30.0@sha256:a2787b40b89a04b33cea95e7304bb2efc0a126c2676d9e6291b59cfb9ced4b88
    container_name: syncthing
    networks:
      - web
      - ldap
      - lan
    environment:
      - PUID={{ ansible_user_uid }}
      - PGID={{ ansible_user_gid }}
      - TZ={{ iana_timezone }}
    labels:
      - "traefik.enable=true"
      - "traefik.docker.network=web"

      - "traefik.http.routers.syncthing.entrypoints=websecure"
      - "traefik.http.routers.syncthing.rule=Host(`sync.{{ ansible_fqdn }}`)"
      - "traefik.http.routers.syncthing.service=syncthing"
      - "traefik.http.services.syncthing.loadbalancer.server.port=8384"

      - traefik.tcp.routers.syncthing.entryPoints=syncthing
      - traefik.tcp.routers.syncthing.rule=HostSNI(`*`)
      - traefik.tcp.routers.syncthing.service=syncthing
      - traefik.tcp.services.syncthing.loadBalancer.server.port=22000
      - traefik.udp.routers.syncthing-udp.entryPoints=syncthing-udp
      - traefik.udp.routers.syncthing-udp.service=syncthing-udp
      - traefik.udp.services.syncthing-udp.loadBalancer.server.port=22000

      - traefik.udp.routers.syncthing-discovery.entryPoints=syncthing-discovery
      - traefik.udp.routers.syncthing-discovery.service=syncthing-discovery
      - traefik.udp.services.syncthing-discovery.loadBalancer.server.port=21027
    volumes:
      - "{{ docker_mounts_directory }}/syncthing/config:/config"
      - /mnt/storage/syncthing/data:/data
    restart: unless-stopped
    mem_limit: 256m
networks:
  web:
    external: true
  ldap:
    external: true
  lan:
    external: true
